### PR TITLE
TMEDIA-346/medium_promo_refactor

### PR DESCRIPTION
**Pull request by farleyawapo**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
Refactor for `medium-promo` and `medium-manual-promo`.
the presentational code for `medium-promo` will likely be able to be used in other blocks as well.

## Jira Ticket
- [TMEDIA-346](https://arcpublishing.atlassian.net/browse/TMEDIA-346)

## Acceptance Criteria
Refactor code between medium promos blocks.
Store presentational logic in shared-styles.
Adjust tests as needed.

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-346/medium_promo_refactor `
2. Run storybook: `npm run storybook`
3. Do a visual test to make `medium-promo` and `medium-manual-promo` blocks look the same locally as they do here: https://canary--5f91e4721ccaba0022a10782.chromatic.com/

## Effect Of Changes
All visuals should stay the same. This is simply a refactor.

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/medium-promo-block/features/medium-promo/default.jsx [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
-  Byline, HeadingSection, PromoDate, PromoDescription, PromoHeadline, PromoImage,
-} from '@wpmedia/shared-styles';
+import { MediumPromoPresentation } from '@wpmedia/shared-styles';
 
 import '@wpmedia/shared-styles/scss/_medium-promo.scss';
 
 const MediumPromoItem = ({ customFields }) => {
-  const { arcSite, isAdmin } = useFusionContext();
-  const { searchableField } = useEditableContent();
+  const { arcSite } = useFusionContext();
```
Comment (matthewroach) [inline blocks/medium-promo-block/features/medium-promo/default.jsx]: Add this to line 96 and pass through as we did for small promo

**Review by matthewroach — CHANGES_REQUESTED** [review]
One small change

File: blocks/shared-styles/_children/promos/medium/index.test.jsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+const { default: mockData } = require('./mock-data');
+
+jest.mock('@wpmedia/engine-theme-sdk', () => ({
+  Image: () => <div />,
+  LazyLoad: ({ children }) => <>{ children }</>,
+  isServerSide: () => true,
+  localizeDateTime: jest.fn(() => new Date().toDateString()),
+}));
+jest.mock('fusion:themes', () => (jest.fn(() => ({}))));
```
Comment (nschubach) [inline blocks/shared-styles/_children/promos/medium/index.test.jsx]:
```suggestion
```
This doesn't seem to be used.

File: blocks/shared-styles/_children/promos/medium/index.test.jsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+jest.mock('fusion:properties', () => (jest.fn(() => ({
+  fallbackImage: 'placeholder.jpg',
+}))));
+jest.mock('fusion:context', () => ({
+  useFusionContext: jest.fn(() => ({
+    arcSite: 'the-sun',
+  })),
+  useComponentContext: jest.fn(() => ({
+    registerSuccessEvent: () => ({}),
```
Comment (nschubach) [inline blocks/shared-styles/_children/promos/medium/index.test.jsx]:
```suggestion
```
Same here.  Not used.

**Review by nschubach — CHANGES_REQUESTED** [review]
Just two minor removals.

File: blocks/shared-styles/_children/promos/medium/index.test.jsx (hunk 2 above)
Comment (farleyawapo) [reply blocks/shared-styles/_children/promos/medium/index.test.jsx]: removed

File: blocks/shared-styles/_children/promos/medium/index.test.jsx (hunk 3 above)
Comment (farleyawapo) [reply blocks/shared-styles/_children/promos/medium/index.test.jsx]: removed

**Review by nschubach — APPROVED** [review]

**Review by matthewroach — APPROVED** [review]
Nice work @farleyawapo 🎉

**Pull request merged**